### PR TITLE
[HttpKernel] Enhance exception message for services not found in dedicated service locators

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes (DX)
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25342, #25196
| License       | MIT
| Doc PR        | -

Misses tests for now.
@weaverryan I may need your help to make the message better if you have any suggestions :)

image

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/HttpKernel.php [hunk 1]
```diff
@@ -150,7 +163,7 @@ private function handleRaw(Request $request, $type = self::MASTER_REQUEST)
         $arguments = $event->getArguments();
 
         // call controller
-        $response = call_user_func_array($controller, $arguments);
+        $response = \call_user_func_array($controller, $arguments);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/HttpKernel.php]: improves perf, and DX, by removing one noisy frame in exception traces

**Comment by curry684** [comment]
I like the detection and how it works out, yet I'm not really sure it solves the DX issue we ran into at #25342.

I mean - the guy just followed the tutorial, derived from `AbstractController` as explicitly advised, and got an inexplicable error. From that background, the new error message, although 10 times more usable to experienced Symfony developers, isn't a single bit clearer to the less seasoned Symfony user, to them it boils down to:
> The service you asked for exists, but not in [something you never asked for] of your [something you never expected or knew to exist]. Did your tutorial really not mention that underneath this base class we use an interface you'll likely never care or learn about with a static function you cannot easily override or extend to not fix this behavior? Because actually you should use DI but we don't have the context at where we throw this exception to be able to tell you that.

And I have no clue how to do this one better either, so I'm also hoping @weaverryan has a brilliant idea on this   😉

**Comment by nicolas-grekas** [comment]
Don't forget that the goal of the message is not to provide full documentation, but mostly to accurately hint into the right direction.
Leaving the details to official doc + Google/SO is already the real solution to solve any errors.

**Comment by nicolas-grekas** [comment]
Oh, and the second part of your comment, about the context, will be fixed on 4.1, once we will hide by default stack frames in vendor (and keep the one from src/ visible.)

**Comment by nicolas-grekas** [comment]
Closing as I'm working on a more local approach.

**Pull request closed without merging**